jkube-kit/config/resource/src/test/java/org/eclipse/jkube/kit/config/resource/ProcessorConfigTest.java

```diff
@@ -110,65 +112,32 @@ public void getValueWithValueInAllShouldReturnValueInProcessorConfig() {
     assertThat(result).isEqualTo("oneInConfig");
   }
 
-  @Test
-  public void getValueWithValueInPropertiesAndDefaultValAndDefaultConfigShouldReturnValueInProperties() {
-    // Given
-    final ProcessorConfig config = new ProcessorConfig();
-    config.getConfig().put("name", Collections.singletonMap("two", "twoInConfig"));
-    final Properties properties = new Properties();
-    properties.put("prefix.name.one", "oneInProp");
-    // When
-    final String result = ProcessorConfig.getConfigValue(
-        config, "name", "prefix", properties, TestConfig.ONE, "defaultValueOne");
-    // Then
-    assertThat(result).isEqualTo("oneInProp");
-  }
-
-  @Test
-  public void getValueWithValueInDefaultValAndDefaultConfigShouldReturnValueInDefaultVal() {
+  @ParameterizedTest(name = "{0}")
+  @MethodSource("getValueTestData")
+  void getValue(String testDesc, String key, TestConfig testConfig, String defaultValue, String expected) {
     // Given
     final ProcessorConfig config = new ProcessorConfig();
     config.getConfig().put("name", Collections.singletonMap("two", "twoInConfig"));
     final Properties properties = new Properties();
-    properties.put("prefix.other-name.one", "oneInProp");
+    properties.put(key, "oneInProp");
     // When
     final String result = ProcessorConfig.getConfigValue(
-        config, "name", "prefix", properties, TestConfig.ONE, "defaultValueOne");
+            config, "name", "prefix", properties, testConfig, defaultValue);
     // Then
-    assertThat(result).isEqualTo("defaultValueOne");
+    assertThat(result).isEqualTo(expected);
   }
 
-  @Test
-  public void getValueWithNoValueAndConfigDefaultShouldReturnValueInConfigDefault() {
-    // Given
-    final ProcessorConfig config = new ProcessorConfig();
-    config.getConfig().put("name", Collections.singletonMap("two", "twoInConfig"));
-    final Properties properties = new Properties();
-    properties.put("prefix.other-name.one", "oneInProp");
-    // When
-    final String result = ProcessorConfig.getConfigValue(
-        config, "name", "prefix", properties, TestConfig.ONE, null);
-    // Then
-    assertThat(result).isEqualTo("configDefaultForOne");
-  }
-
-  @Test
-  public void getValueWithNoValueShouldReturnNull() {
-    // Given
-    final ProcessorConfig config = new ProcessorConfig();
-    config.getConfig().put("name", Collections.singletonMap("two", "twoInConfig"));
-    final Properties properties = new Properties();
-    properties.put("prefix.other-name.one", "oneInProp");
-    // When
-    final String result = ProcessorConfig.getConfigValue(
-        config, "name", "prefix", properties, TestConfig.NO_DEFAULT, null);
-    // Then
-    assertThat(result).isNull();
+  public static Stream<Arguments> getValueTestData() {
+    return Stream.of(
+            Arguments.of("withValueInPropertiesAndDefaultValAndDefaultConfigShouldReturnValueInProperties", "prefix.name.one", TestConfig.ONE, "defaultValueOne", "oneInProp"),
```

Comment:
Since you are providing the test descriptions as strings, maybe it's a good time to make them a little bit more readable ;)